eng/pipelines/libraries/run-test-job.yml

```diff
@@ -64,14 +71,85 @@ jobs:
           - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
         - ${{ if ne(parameters.liveRuntimeBuildConfig, '') }}:
           - ${{ format('{0}_{1}_product_build_{2}{3}_{4}_{5}', parameters.runtimeFlavor, parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveRuntimeBuildConfig) }}
+      # SuperPMI collection needs to run mcs.exe on the AzDO machine. Assume that's an x64 machine, and download an x64 product build if needed.
+      - ${{ if and(eq(parameters.SuperPmiCollect, true), ne(parameters.archType, 'x64')) }}:
+          - ${{ format('{0}_{1}_product_build_{2}{3}_{4}_{5}', 'coreclr', '', parameters.osGroup, parameters.osSubgroup, 'x64', parameters.liveRuntimeBuildConfig) }}
 
       variables:
+        # Convenience variables
+        - name: buildConfig
+          value: ${{ parameters.buildConfig }}
+        - name: archType
+          value: ${{ parameters.archType }}
+        - name: osGroup
+          value: ${{ parameters.osGroup }}
+        - name: osSubgroup
+          value: ${{ parameters.osSubgroup }}
+        - name: buildConfigUpper
+          ${{ if eq(parameters.buildConfig, 'debug') }}:
+            value: 'Debug'
+          ${{ if eq(parameters.buildConfig, 'release') }}:
+            value: 'Release'
+          ${{ if eq(parameters.buildConfig, 'checked') }}:
+            value: 'Checked'
```

Comment:
I'm not exactly happy about defining these variables here. We don't need any of those in src/libraries and some of those might already be defined. I would prefer if you find a way to avoid them.

In general, as the SuperPMI logic is completely optional it would be good to not increase the YML complexity. Can you move this into a template?

Reply:
It's a little unfortunate these aren't already defined at a higher level (base-job.yml?) -- they are available in the coreclr side.